simulations/src/overlay/overlay_info.rs

```diff
@@ -0,0 +1,181 @@
+use consensus_engine::{CommitteeId, NodeId, Overlay};
+use serde::Serialize;
+use std::collections::{BTreeSet, HashMap, VecDeque};
+
+pub type Blake2bU32 = blake2::Blake2b<digest::typenum::U32>;
+
+#[derive(Debug, Serialize)]
+pub struct OverlayInfo {
+    pub committees: BTreeSet<CommitteeId>,
+    pub committee_sizes: HashMap<CommitteeId, usize>,
+    pub edges: Vec<(CommitteeId, CommitteeId)>,
+    pub next_leader: NodeId,
+    pub root_id: CommitteeId,
+}
+
+pub trait OverlayInfoExt {
```

Comment:
You can make it require `Overlay` as well:
```rust
pub trait OverlayInfoExt : Overlay
```

Reply:
I'm leaning towards leaving it as is for now, until it's required. As I understand, the implementation definition (`impl<T: Overlay> OverlayInfoExt for T {`) wouldn't change with `pub trait OverlayInfoExt : Overlay`?